site/_posts/2021-04-01-why-user-multi-factor-authentication.md

```diff
@@ -0,0 +1,129 @@
+
+When an action like this is requested, the system needs certainty about the requester. These actions can be, and often are, legitimate. But they can also be used by someone with illicit access to the user's account to solidify the takeover.
+
+By using step up auth, you can help mitigate the damage of a compromised account. An attacker may be able to access the account and read data from it, but if they don't have the additional factor, they won't be able to undertake actions which can cause more damage.
+
+### Legal or organizational policies
+
+If your application is used by certain organizations or stores personally identifiable information, you may need to require multi-factor authentication for some or all accounts. As part of the NIST risk management framework, for example, Authenticator Assurance Level 2 requires: ["proof of possession and control of two different authentication factors is required..."](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-63-3.pdf). 
+
+Sometimes an MFA requirement is not explicit, however. If you are seeking SOC2 certification, you will need to implement MFA, even though multi-factor authentication is never mentioned in the [SOC "Trust Services Criteria"](https://www.aicpa.org/content/dam/aicpa/interestareas/frc/assuranceadvisoryservices/downloadabledocuments/trust-services-criteria.pdf).
+
+From Section CC6.1 of the SOC document: "Persons, infrastructure, and software are identified and authenticated prior to accessing information assets, whether locally or remotely". The document omits specific implementation details. Talk to your SOC2 auditor about required controls.
+
+### When actions look suspicious
+
+An auth system has a lot of information about who is logging in. During authentication, credentials are supplied; the end result is the answer to this question: "Is the user behind this authentication request who they are claiming to be?" 
+
+Some data, such as the username and password, is explicitly provided to the system. But there is also interesting implicit data available such as:
+
+* The date and time of access
+* Connection information such as IP address and user agent
+* How many times the user has logged in recently
+* Has this device been used to access this system before
+
+Such implicit data can help determine if the authentication request is legitimate. For instance, if a user accesses a system from the USA and 24 hours later there is another request from Germany with the same credentials, is the latter valid? Could be, but it could also be that the request from Germany is due to a compromised account. Requiring MFA at such a time allows you to apply extra scrutiny to strange requests.
+
+Another example where you might want additional authentication factors is when a user logs in from a new device. Popular SaaS applications such as Google's GSuite and Mailchimp can be configured to require MFA for only the initial authentication from a new device.
```

Comment:
Possibly not just a device (might not be important)

Another example where you might want additional authentication factors is when a user logs in from a new device (or IP, if through a new proxy).